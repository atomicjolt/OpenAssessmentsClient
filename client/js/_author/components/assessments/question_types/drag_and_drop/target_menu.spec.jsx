import React          from 'react';
import ReactDOM       from 'react-dom';
import { shallow }    from 'enzyme';
import renderer       from 'react-test-renderer';
import TargetMenu     from './target_menu';

describe('target_menu component', () => {
  let props;
  let result;
  let uploadedMedia;
  let zoneNew;

  beforeEach(() => {
    uploadedMedia = false;
    zoneNew = false;
    props = {
      uploadMedia: () => {uploadedMedia = true},
      newZone: () => {zoneNew = true},
      images: {},
      loadingMedia: false,
      hasTarget: false,
    };
    result = shallow(<TargetMenu {...props} />);
  });

  it('renders the component as a snapshot', () => {
    expect(result).toMatchSnapshot();
  });

  it('handles the uploadMedia prop call', () => {
    expect(uploadedMedia).toBeFalsy();
    result.instance().replaceImage();
    expect(uploadedMedia).toBeTruthy();
  });

  it('handles the newZone prop call', () => {
    expect(zoneNew).toBeFalsy();
    result.instance().addByRegion();
    expect(zoneNew).toBeTruthy();
  });

  it('handles setting state', () => {
    expect(result.instance().state.add).toBe(null);
    expect(result.instance().state.showModal).toBeFalsy();
    result.find('button').simulate('click');
    expect(result.instance().state.showModal).toBeTruthy();
  });
});
