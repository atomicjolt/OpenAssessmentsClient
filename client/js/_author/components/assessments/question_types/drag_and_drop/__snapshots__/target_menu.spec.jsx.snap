// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`target_menu component renders the component as a snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="au-o-item__top"
>
    <div
        className="au-o-left"
    >
        <div
            className="au-c-question__type"
        >
            Target Image
        </div>
    </div>
    <div
        className="au-o-right"
    >
        <button
            className="au-c-btn au-c-btn--sm au-c-btn--gray"
            onClick={[Function]}
        >
            Add Image
        </button>
        <addZoneDropdown
            active={false}
            addByImage={[Function]}
            addByRegion={[Function]}
            text="Add Snap Zone"
            toggle={[Function]}
        />
        <addZoneDropdown
            active={false}
            addByImage={[Function]}
            addByRegion={[Function]}
            text="Add Drop Zone"
            toggle={[Function]}
        />
    </div>
    <EditorUploadModal
        closeModal={[Function]}
        insertMedia={[Function]}
        isOpen={false}
        loading={false}
        media={Object {}}
        mediaType="img"
        uploadOnly={false}
    />
</div>,
  "nodes": Array [
    <div
      className="au-o-item__top"
>
      <div
            className="au-o-left"
      >
            <div
                  className="au-c-question__type"
            >
                  Target Image
            </div>
      </div>
      <div
            className="au-o-right"
      >
            <button
                  className="au-c-btn au-c-btn--sm au-c-btn--gray"
                  onClick={[Function]}
            >
                  Add Image
            </button>
            <addZoneDropdown
                  active={false}
                  addByImage={[Function]}
                  addByRegion={[Function]}
                  text="Add Snap Zone"
                  toggle={[Function]}
            />
            <addZoneDropdown
                  active={false}
                  addByImage={[Function]}
                  addByRegion={[Function]}
                  text="Add Drop Zone"
                  toggle={[Function]}
            />
      </div>
      <EditorUploadModal
            closeModal={[Function]}
            insertMedia={[Function]}
            isOpen={false}
            loading={false}
            media={Object {}}
            mediaType="img"
            uploadOnly={false}
      />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <TargetMenu
        hasTarget={false}
        images={Object {}}
        loadingMedia={false}
        localizeStrings={[Function]}
        newZone={[Function]}
        uploadMedia={[Function]}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": TargetMenu {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "hasTarget": false,
          "images": Object {},
          "loadingMedia": false,
          "localizeStrings": [Function],
          "newZone": [Function],
          "uploadMedia": [Function],
        },
        "refs": Object {},
        "state": Object {
          "add": null,
          "showModal": false,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="au-o-item__top"
>
          <div
                    className="au-o-left"
          >
                    <div
                              className="au-c-question__type"
                    >
                              Target Image
                    </div>
          </div>
          <div
                    className="au-o-right"
          >
                    <button
                              className="au-c-btn au-c-btn--sm au-c-btn--gray"
                              onClick={[Function]}
                    >
                              Add Image
                    </button>
                    <addZoneDropdown
                              active={false}
                              addByImage={[Function]}
                              addByRegion={[Function]}
                              text="Add Snap Zone"
                              toggle={[Function]}
                    />
                    <addZoneDropdown
                              active={false}
                              addByImage={[Function]}
                              addByRegion={[Function]}
                              text="Add Drop Zone"
                              toggle={[Function]}
                    />
          </div>
          <EditorUploadModal
                    closeModal={[Function]}
                    insertMedia={[Function]}
                    isOpen={false}
                    loading={false}
                    media={Object {}}
                    mediaType="img"
                    uploadOnly={false}
          />
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="au-o-item__top"
>
          <div
                    className="au-o-left"
          >
                    <div
                              className="au-c-question__type"
                    >
                              Target Image
                    </div>
          </div>
          <div
                    className="au-o-right"
          >
                    <button
                              className="au-c-btn au-c-btn--sm au-c-btn--gray"
                              onClick={[Function]}
                    >
                              Add Image
                    </button>
                    <addZoneDropdown
                              active={false}
                              addByImage={[Function]}
                              addByRegion={[Function]}
                              text="Add Snap Zone"
                              toggle={[Function]}
                    />
                    <addZoneDropdown
                              active={false}
                              addByImage={[Function]}
                              addByRegion={[Function]}
                              text="Add Drop Zone"
                              toggle={[Function]}
                    />
          </div>
          <EditorUploadModal
                    closeModal={[Function]}
                    insertMedia={[Function]}
                    isOpen={false}
                    loading={false}
                    media={Object {}}
                    mediaType="img"
                    uploadOnly={false}
          />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <TargetMenu
    hasTarget={false}
    images={Object {}}
    loadingMedia={false}
    localizeStrings={[Function]}
    newZone={[Function]}
    uploadMedia={[Function]}
/>,
}
`;
